File: backend/core-service/src/Fix.Application/UseCases/Policies/Dtos/PolicyDto.cs
using Fix.Domain.AggregateRoots.Policies;

namespace Fix.Application.Policies.Dtos;

/// <summary>Política-mãe completa (parâmetros, eixos, bandas, instrumentos e histórico de versões).</summary>
public sealed record PolicyDto(
    Guid Id,
    string Code,
    string Title,
    string Version,
    string? Description,
    PolicyStatus Status,
    DateOnly ValidFrom,
    DateOnly? ValidTo,
    string? ApprovalRecord,
    DateOnly? ApprovedOn,
    PolicyLimitsDto Limits,
    IReadOnlyList<PolicyAxisDto> Axes,
    IReadOnlyList<CoverageBandDto> Bands,
    IReadOnlyList<PolicyInstrumentDto> Instruments,
    IReadOnlyList<PolicyVersionDto> Versions);

File: frontend/bff/src/application/policies/dtos/policy.dto.ts
import type { CoverageBandDto } from './coverage-band.dto.js';
import type { PolicyAxisDto } from './policy-axis.dto.js';
import type { PolicyInstrumentDto } from './policy-instrument.dto.js';
import type { PolicyLimitsDto } from './policy-limits.dto.js';
import type { PolicyStatus } from '../../../cross-cutting/enums/policy-status.js';
import type { PolicyVersionDto } from './policy-version.dto.js';

export interface PolicyDto {
  id: string;
  code: string;
  title: string;
  version: string;
  description: string | null;
  status: PolicyStatus;
  validFrom: string;
  validTo: string | null;
  approvalRecord: string | null;
  approvedOn: string | null;
  limits: PolicyLimitsDto;
  axes: PolicyAxisDto[];
  bands: CoverageBandDto[];
  instruments: PolicyInstrumentDto[];
  versions: PolicyVersionDto[];
}
